### PR TITLE
Add SDK and platform paths to plugin executables

### PR DESCRIPTION
**Pull request by DougGregor**

### CODE REVIEW

**Comment by DougGregor** [comment]
@swift-ci please test

File: Sources/SwiftDriver/Jobs/FrontendJobHelpers.swift [hunk 1]
```diff
@@ -27,6 +27,26 @@ fileprivate func shouldColorDiagnostics() -> Bool {
   return TerminalController.isTTY(stderrStream)
 }
 
+extension VirtualPath {
+  // Given a virtual path pointing into a toolchain/SDK/platform, produce the
+  // path to `swift-plugin-server`.
+  fileprivate var pluginServerPath: VirtualPath {
+    self.appending(components: "usr", "swift-plugin-server")
```
Comment (bnbarham) [inline Sources/SwiftDriver/Jobs/FrontendJobHelpers.swift:34]: `usr/bin`?

Comment (bnbarham) [reply Sources/SwiftDriver/Jobs/FrontendJobHelpers.swift:34]: Though seems like the rest don't have `usr`, so is this actually intended to be `bin`?

File: Tests/SwiftDriverTests/SwiftDriverTests.swift [hunk 2]
```diff
@@ -6798,14 +6798,16 @@ final class SwiftDriverTests: XCTestCase {
...
+    let sdkRoot = testInputsPath.appending(component: "SDKChecks").appending(component: "iPhoneOS.sdk")
+    var driver = try Driver(args: ["swiftc", "-typecheck", "foo.swift", "-sdk", VirtualPath.absolute(sdkRoot).name])
+    guard driver.isFrontendArgSupported(.pluginPath) && driver.isFrontendArgSupported(.externalPluginPath) else {
       return
     }
 
     let jobs = try driver.planBuild().removingAutolinkExtractJobs()
     XCTAssertEqual(jobs.count, 1)
     let job = jobs.first!
+    XCTAssertTrue(job.commandLine.contains(.flag("-external-plugin-path")))
```
Comment (bnbarham) [inline Tests/SwiftDriverTests/SwiftDriverTests.swift:6810]: Might be worth checking the path here 😅

File: Sources/SwiftDriver/Jobs/FrontendJobHelpers.swift [hunk 3]
```diff
@@ -261,18 +281,46 @@ extension Driver {
       try commandLine.appendLast(.emitMacroExpansionFiles, from: &parsedOptions)
     }
 
-    if isFrontendArgSupported(.pluginPath) {
+    // Emit user-provided plugin paths, in order.
+    if isFrontendArgSupported(.externalPluginPath) {
+      try commandLine.appendAll(.pluginPath, .externalPluginPath, from: &parsedOptions)
```
Comment (rintaro) [inline Sources/SwiftDriver/Jobs/FrontendJobHelpers.swift:286]: Aren't we passing `load-plugin-library` and `load-plugin-executable`? 🤔

File: Sources/SwiftDriver/Jobs/FrontendJobHelpers.swift (hunk 1 above)
Comment (DougGregor) [reply Sources/SwiftDriver/Jobs/FrontendJobHelpers.swift:34]: You're absolutely right.

File: Sources/SwiftDriver/Jobs/FrontendJobHelpers.swift (hunk 3 above)
Comment (DougGregor) [reply Sources/SwiftDriver/Jobs/FrontendJobHelpers.swift:286]: I think those also need to be handled in this same group, yes.

Comment (DougGregor) [reply Sources/SwiftDriver/Jobs/FrontendJobHelpers.swift:286]: Woah, those aren't being passed at all!

**Comment by DougGregor** [comment]
@swift-ci please test

**Comment by DougGregor** [comment]
@swift-ci please test Windows

**Review by rintaro — APPROVED** [review]

**Pull request merged**